crates/network-scanner-net/src/tokio_raw_socket.rs

```diff
@@ -12,6 +15,14 @@ pub struct TokioRawSocket {
     socket: Arc<Mutex<socket2::Socket>>,
 }
 
+macro_rules! lock_socket {
+    ($socket:expr) => {
+        $socket
+            .lock()
+            .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, format!("failed to lock socket: {}", e)))?
+    };
+}
```

Comment:
suggestion: We typically don’t go to such extends to handle poisoning.

```rust
my_mutex.lock().unwrap();
```

Is typically enough. Poisoning will happen when another thread holding the mutex panicked, so it’s generally fine to propagate the panic.

Alternatively, `parking_lot` is providing a Mutex without poisoning (that we already use in Devolutions Gateway).

```
my_mutex.lock();
```